`mockk-android:1.12.7` always fails with `'dispatcher.jar' not found`

### Prerequisites

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

### Expected Behavior

Not fails.

### Current Behavior

The instrumentation test fails with the following stack trace.

#### Context

* MockK version: 1.12.7

#### Stack trace

```
java.lang.NoClassDefFoundError: io.mockk.impl.JvmMockKGateway
at MyInstrumentationTest.tearDown(SettingsFragmentTest.kt:803)
... 37 trimmed
Caused by: java.lang.ExceptionInInitializerError
at MyInstrumentationTest.setUp(SettingsFragmentTest.kt:668)
at java.lang.reflect.Method.invoke(Native Method)
at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
at androidx.test.internal.runner.junit4.statement.RunBefores.evaluate(RunBefores.java:76)
at androidx.test.internal.runner.junit4.statement.RunAfters.evaluate(RunAfters.java:61)
... 33 more
Caused by: io.mockk.proxy.MockKAgentException: 'dispatcher.jar' not found
at io.mockk.proxy.android.AndroidMockKAgentFactory.init(AndroidMockKAgentFactory.kt:47)
at io.mockk.impl.JvmMockKGateway.<init>(JvmMockKGateway.kt:46)
at io.mockk.impl.JvmMockKGateway.<clinit>(JvmMockKGateway.kt:186)
... 40 more
```

#### Minimal reproducible code (the gist of this issue)

```kotlin
dependencies {
    androidTestImplementation("io.mockk:mockk:1.12.7")
    androidTestImplementation("io.mockk:mockk-android:1.12.7")
    // Adding the following does not change the behavior
//    androidTestImplementation("io.mockk:mockk-agent:1.12.7")
//    androidTestImplementation("io.mockk:mockk-agent-android:1.12.7")
}
```

--- a/modules/mockk-agent-android/build.gradle.kts
+++ b/modules/mockk-agent-android/build.gradle.kts
@@ -23,7 +23,7 @@ android {
     sourceSets {
         named("main").configure {
             resources {
-                srcDirs(dispatcherJarResPath)
+                srcDirs({ packageDispatcherJar.map { it.destinationDirectory } })
             }
         }
     }
@@ -60,13 +60,11 @@ dependencies {
     androidClassesDex(projects.modules.mockkAgentAndroidDispatcher)
 }
 
-val dispatcherJarResPath: Provider<Directory> = layout.buildDirectory.dir("generated/dispatcher-jar")
-
 val packageDispatcherJar by tasks.registering(Jar::class) {
     group = LifecycleBasePlugin.BUILD_GROUP
     from(androidClassesDex.asFileTree)
     archiveFileName.set("dispatcher.jar")
-    destinationDirectory.set(dispatcherJarResPath)
+    destinationDirectory.set(temporaryDir)
 }
 
 tasks.preBuild {

